superset/viz.py

```diff
@@ -1949,17 +1949,23 @@ def tupleify(s):
 
             df[key] = df[spatial.get('lonlatCol')].apply(tupleify)
 
-            if spatial.get('reverseCheckbox'):
-                df[key] = [
-                    tuple(reversed(o)) if isinstance(o, (list, tuple)) else (0, 0)
-                    for o in df[key]
-                ]
             del df[spatial.get('lonlatCol')]
         elif spatial.get('type') == 'geohash':
-            latlong = df[spatial.get('geohashCol')].map(geohash.decode)
+            def try_geohash(h):
+                try:
+                    latlng = geohash.decode(h)
+                except Exception as e:
+                    latlng = (0, 0)
```

Comment:
I'm worried about returning valid values where we have bad data. What happens when a user filters on region (say, SFO), and then tries to average lat and lon to look at some "average region" — these invalid data points would skew the result. It's an unlikely but not impossible scenario, and it would be hard to troubleshoot.

Also, how does this play with your auto-zoom feature? Is the viz going to zoom out in order to include this point when bad data is present?

Is there any way we can be explicit and return `None` here, and deal with it higher in the stack?

Reply:
I agree that this should be handled better. I think the function I use for autozoom (and perhaps other layers of the code) don't play well with `None`. I was thinking about filtering out the `Nones` out right there but that should probably warn the user if rows are getting removed.

Probably the ideal would be to carry the `None` out to the frontend and warn over there using `notify.warn` that some rows got removed.